### DISCUSSION TITLE
Use uint64 to hold queue size in message stats ?

### DISCUSSION THREAD

**Discussion by sebastian-nagel** [crawler-commons/url-frontier#10] [General]
The number of queued URLs in the message stats is hold in a uint32 type (see [urlfrontier.proto](/crawler-commons/url-frontier/blob/c9dcb3aeb14c0c5e92ffcc75e413c873e76a0a77/API/urlfrontier.proto#L51)) which naturally limits the max. number of URLs to 2^32 resp. 2^31 if there are no unsigned integer types in the target programming language (eg. Java).

What about using uint64 instead?
- because protobuf uses a [varint encoding](https://developers.google.com/protocol-buffers/docs/encoding#varints] this causes no overhead when sending message stats
- in doubt, I would assume that there are distributed crawlers which can handle more than 2 billions queued URL. Of course, there is no need for a queue implementation to handle such large queues.
- open question: use uint64 for all values, including per-queue size, number of queues? Again, in doubt, why not?
  - a sitemap index may list 2.5 billion URLs (50k*50k)
  - there are for sure more than 2 billion host names in the web
- btw., [ListQueues](https://github.com/crawler-commons/url-frontier/blob/c9dcb3aeb14c0c5e92ffcc75e413c873e76a0a77/API/urlfrontier.proto#L27) takes uint64 as parameter

**Comment by jnioche** [comment]
thanks @sebastian-nagel
for Stats: yes they could all be uint64 - apart from maybe from _inprocess_ but that wouldn't matter if it did

**Comment by jnioche** [comment]
ListQueues probably doesn;t need to be a uint64 but it doesn' tmatter really

**Reply by sebastian-nagel** [reply]
Indeed, you hardly want to see so many queues. But yes, no matter.

**Comment by jnioche** [comment]
@sebastian-nagel implemented in a21656ad9aea8c3abc0a92f30d837c530f83dd79
